### PR TITLE
Update NMEA satellites count (sat inuse, sat inview)

### PR DESCRIPTION
**Pull request by bettellam**
With this proposed NMEA code change, the count:
- number of satellites in use;
- number of satellists in the view of the GNSS receiver
is resolved.
**
The count of satellites in use was previously set up to read GPGGA / GNGGA messages (with a maximum result of 12 satellites), with this modification the count has been moved to GPGSA / GNGSA messages.
https://receiverhelp.trimble.com/alloy-gnss/en-us/NMEA-0183messages_gsa.html
**
As for satellites in view to GNSS receiver, I changed the routine related to GxGSV messages, in particular to previous GPGSV / GNGSV messages, they were added:
GLGSV (Glonass),
GBGSV (Beidou),
GAGSV (Galileo).
https://receiverhelp.trimble.com/alloy-gnss/en-us/NMEA-0183messages_gsv.html
**
Since the GxGSA and GxGSV messages are multi-line in the NMEA sentence, the GxVTG message is used as a "clock" to reset the satellite counts and associated values.
**
I updated the GPS Tools widget (it should be called: GNSS Tools) and reactivated: SkyPlot satellites
**
The tests were performed with U-Blox ZED-F9P GNSS receiver with excellent results comparable with other GNSS applications.
I hope the QField version can benefit from this update.
**
I'm not a programmer, I don't know the GitHub platform, I don't know C ++ programming and QGis code, over the years I have learned to read the code a bit, I don't know many other things, I don't know if I did this in correct way, I hope to receive support for this work to go to "Good ending !!"
Thank you
QGIS_NMEA_sat_used_sat_onview_1
------------
QGIS_NMEA_sat_used_sat_onview_2

------------
QGIS_NMEA_sat_used_sat_onview_3

### CODE REVIEW

**Comment by nyalldawson** [comment]
Can you please remove all the unrelated changes here? (Eg the new codeql workflow, WITH_QWTPOLAR changes, etc)

**Comment by bettellam** [comment]
> Can you please remove all the unrelated changes here? (Eg the new codeql workflow, WITH_QWTPOLAR changes, etc)

Ok, can I leave the indication on satellites: in use / in view?

**Comment by bettellam** [comment]
@nyalldawson I have the changes I made in:
CMakeLists.txt (1 change);
qgsgpsinformationwidget.cpp (2 changes)
The change to line 976 could be left for more information to user !!!

**Comment by bettellam** [comment]
@nyalldawson 
in all tests that I have run CodeQL has never been successful !!!
I do not know what I have to do!!!

**Comment by m-kuhn** [comment]
@bettellam the codeql workflow has been added in this pull request. Remove the file again.
Please also set up a local development environment to test your code changes.

**Comment by bettellam** [comment]
> 

@m-kuhn I'm sorry it's the first time I've done these things !!
Which file should I remove?

File: src/core/gps/qgsnmeaconnection.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 #include "qgsnmeaconnection.h"
 #include "qgslogger.h"
+#include "qgsgpsconnection.h"
```
Comment (agiudiceandrea) [inline src/core/gps/qgsnmeaconnection.cpp:20]:
```suggestion
```

**Comment by agiudiceandrea** [comment]
> Which file should I remove?

.github/workflows/codeql-analysis.yml

File: external/nmea/parse.c [hunk 2]
```diff
@@ -390,7 +390,9 @@ int nmea_parse_GPGSV( const char *buff, int buff_sz, nmeaGPGSV *pack )
     return 0;
   }
 
-  if ( type != 'P' && type != 'N' )
+  if ( type != 'P' && type != 'N' && type != 'L' && type != 'A' && type != 'B' )
+  //if ( type != 'P' && type != 'N' && type != 'L' && type != 'A' && type != 'B' && type != 'Q' )
```
Comment (agiudiceandrea) [inline external/nmea/parse.c:394]: I think you could remove the lines you have commented out (here and elsewhere).

Comment (bettellam) [reply external/nmea/parse.c]: OK fatto

File: src/core/gps/qgsnmeaconnection.cpp (hunk 1 above)
Comment (bettellam) [reply src/core/gps/qgsnmeaconnection.cpp:20]: I added #include "qgsgpsconnection.h" because
"MinGW64 Windows Build" failed

**Comment by agiudiceandrea** [comment]
@bettellam why did you delete qgsnmeaconnection.cpp? This PR is now not suitable to be merged.

**Comment by bettellam** [comment]
> @bettellam why did you delete qgsnmeaconnection.cpp? This PR is now not suitable to be merged.
@agiudiceandrea 
I am a fool.
I was thinking of deleting an old file.

**Comment by m-kuhn** [comment]
@bettellam I would suggest to work locally instead of the web editor. That should make it easier.

https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/adding-and-cloning-repositories/cloning-and-forking-repositories-from-github-desktop

**Comment by bettellam** [comment]
@agiudiceandrea @m-kuhn @nyalldawson I can recover somehow
..... or must close the "Pull Request" ???

**Comment by m-kuhn** [comment]
It's probably easiest to close it and start a new one (with a local editor) with all the learnings taken from this one.
Enjoy being a grandfather :-)

**Comment by bettellam** [comment]
@agiudiceandrea @m-kuhn 
I follow your advice, close "Pull Request"

**Pull request closed without merging**